go/controller/api/v1alpha1/deploymentsettings_types.go

```diff
@@ -691,3 +700,79 @@ func (in *VertexSettings) ServiceAccountJSON(ctx context.Context, c client.Clien
 	res, err := utils.GetSecretKey(ctx, c, in.ServiceAccountJsonSecretRef, namespace)
 	return lo.ToPtr(res), err
 }
+
+type VectorStore struct {
+	// +kubebuilder:default=false
+	// +kubebuilder:validation:Optional
+	Enabled *bool `json:"enabled,omitempty"`
+
+	// +kubebuilder:validation:Enum=ELASTIC
+	// +kubebuilder:validation:Optional
+	VectorStore *console.VectorStore `json:"vectorStore,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	Elastic *ElasticsearchConnectionSettings `json:"elastic,omitempty"`
+}
+
+func (in *VectorStore) Attributes(ctx context.Context, c client.Client, namespace string) (*console.VectorStoreAttributes, error) {
+	if in == nil {
+		return nil, nil
+	}
+
+	if lo.FromPtr(in.Enabled) && in.VectorStore == nil {
+		return nil, fmt.Errorf("vector store type has to be set if it is enabled")
+	}
+
+	attr := &console.VectorStoreAttributes{
+		Enabled: in.Enabled,
+		Store:   in.VectorStore,
+	}
+
+	switch *in.VectorStore {
+	case console.VectorStoreElastic:
+		if in.Elastic == nil {
+			return nil, fmt.Errorf("must provide elastic configuration to set the provider to ELASTIC")
+		}
+
+		password, err := in.Elastic.Password(ctx, c, namespace)
+		if err != nil {
+			return nil, err
+		}
+
+		attr.Elastic = &console.ElasticsearchConnectionAttributes{
+			Host:     in.Elastic.Host,
+			Index:    in.Elastic.Index,
+			User:     in.Elastic.User,
+			Password: password,
+		}
+	}
+
+	return attr, nil
+}
+
+type ElasticsearchConnectionSettings struct {
+	// +kubebuilder:validation:Required
+	Host string `json:"host"`
+
+	// +kubebuilder:validation:Required
+	Index string `json:"index"`
+
+	// +kubebuilder:validation:Optional
+	User *string `json:"user,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	PasswordSecretRef *corev1.SecretKeySelector `json:"passwordSecretRef,omitempty"`
+}
+
+func (in *ElasticsearchConnectionSettings) Password(ctx context.Context, c client.Client, namespace string) (*string, error) {
+	if in == nil {
+		return nil, fmt.Errorf("configured elastic settings cannot be nil")
```

Comment:
we need to make sure that error is able to be backoffed properly by the controller when the secret doesn't yet exist

Reply:
Can you review controller improvements that we did with @zreigz? It would make it easier. Otherwise I can just take some bits from that PR.

Reply:
I've integrated part of the changes from https://github.com/pluralsh/console/pull/1893. Now, `not found` error will be propagated to the `Reconcile` function and handled in `handleRequeue` function.

Reply:
oh my fault, yea approved that pr